src/edu/duke/cs/osprey/structure/AtomConnectivity.java

```diff
@@ -224,7 +224,10 @@ public AtomPairs getAtomPairs(Residue res1, Residue res2) {
 	}
 
 	private boolean isInterResBondedForward(Residue res1, Residue res2) {
-		return res1.template.interResBonding.getClass() == res2.template.interResBonding.getClass()
+		var class1 = res1.template.interResBonding.getClass();
```

Comment:
This seems unnecessary, but it doesn't hurt anything I guess.

Reply:
This function was actually the source of the issue! The cystine template is a subclass of the peptide template, and so `==` is false.  So they didn't have a peptide bond. But the cystine template is assignable to the peptide template since it's a subclass.

Reply:
Oh really? Sorry, I hadn't noticed there was a non-trivial inheritance hierarchy here. I usually avoid those because they can cause all kinds of unforseen problems later on. I guess Mark wrote that one. =P

Carry on. =)